src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -389,7 +390,37 @@ public override async Task<bool> AreSynchronized(Uri firstResourceUri, Uri secon
             {
                 if (await source.ExistsAsync())
                 {
-                    return !string.IsNullOrEmpty(source.Properties.ContentMD5) && source.Properties.ContentMD5 == destination.Properties.ContentMD5;
+                    var sourceBlobMetadata = source.Metadata;
+                    var destinationBlobMetadata = destination.Metadata;
+                    if (sourceBlobMetadata == null || destinationBlobMetadata == null)
+                    {
+                        return false;
+                    }
+
+                    var sourceBlobHasSha512Hash = sourceBlobMetadata.TryGetValue(Sha512HashAlgorithmId, out var sourceBlobSha512Hash);
+                    var destinationBlobHasSha512Hash = destinationBlobMetadata.TryGetValue(Sha512HashAlgorithmId, out var destinationBlobSha512Hash);
+                    if (!sourceBlobHasSha512Hash)
+                    {
+                        Trace.WriteLine(string.Format("The source blob ({0}) doesn't have the Sha512 hash.", source.Uri.ToString()));
+                    }
+                    if (!destinationBlobHasSha512Hash)
+                    {
+                        Trace.WriteLine(string.Format("The destination blob ({0}) doesn't have the Sha512 hash.", destination.Uri.ToString()));
```

Comment:
nit: `SHA512` instead of `Sha512`

Reply:
nit : you can return false here and at line 405. In that case you would not need the `if (sourceBlobHasSha512Hash && destinationBlobHasSha512Hash)` branch below

Reply:
Yeah, the reason that I prefer not to return earlier is that if both of source and destination have no SHA512 hash, we want to log both of them rather than just the source one and return immediately.